src/storage/src/table/batch_table/storage_table.rs

```diff
@@ -433,8 +426,10 @@ impl<S: StateStore> StorageTable<S> {
             .into_iter()
             .map(|index| self.pk_indices[index])
             .collect_vec();
-        let prefix_hint = if self.dist_key_indices.is_empty()
-            || self.dist_key_indices != pk_prefix_indices
+        let dist_key_hint = if self.dist_key_indices.is_empty()
+            || self.distribution_key_start_index_in_pk.is_none()
+            || self.dist_key_indices.len() + self.distribution_key_start_index_in_pk.unwrap()
+                > pk_prefix.len()
```

Comment:
Using `let Some` seems more natural to me and we can avoid unwrap in multiple places.